### ISSUE TITLE
KuCoin : Could not fetch new trading fees from Kucoin. Filling log file and too many API calls

### ISSUE THREAD

**Issue by globeone** [hummingbot/hummingbot#5273]
**Describe the bug**
Upgraded two robots to Ubuntu 22.04 LTS and Hummingbot 1.3.0 , when I start the robot, the robot creates trades on KuCoin but then starts spamming the log file and API every second with the message "WARNING - Could not fetch new trading fees from Kucoin. Check network connection.". It says "Check network connection" but the network connection is working fine. Then a few lines up it says invalid KC-API-SIGN, yet the robot successfully made orders on KuCoin.
When left running, this behavior leads to a full log file and a temporary ban on the KuCoin API.

Other projects are having similar issues it looks like: 
https://github.com/mscheetz/KuCoinApi.Net/issues/13
https://github.com/mscheetz/KuCoinApi.Net/issues/10

**Steps To Reproduce**

- Set up a Docker robot with
- Ubuntu 22.04 LTS VPS
- Snap Docker
- Hummingbot 1.3.0
- API Connect to KuCoin
- Liquidity Mining strategy
- 5 coinpairs
- start
- observe that the robot creates 5 to 10 orders
- observe the robot throwing the warning message
- observe the log file filling up rapidly
- let run for 24 hours
- observe a rate limit temporary ban on the KuCoin API

**Screenshots**
![image](https://user-images.githubusercontent.com/10052587/166113851-bbed3e27-a1f8-4228-9fce-003973c8f6d4.png)

**Release version**
1.3.0

**Attachments**
// Please attach your config file and log file to help us debug. It would be difficult for us to help you without those! 

`Traceback (most recent call last):
  File "/home/hummingbot/hummingbot/connector/exchange/kucoin/kucoin_exchange.py", line 767, in _trading_fees_polling_loop
    await self._update_trading_fees()
  File "/home/hummingbot/hummingbot/connector/exchange/kucoin/kucoin_exchange.py", line 817, in _update_trading_fees
    resp = await self._api_request(
  File "/home/hummingbot/hummingbot/connector/exchange/kucoin/kucoin_exchange.py", line 946, in _api_request
    return await web_utils.api_request(
  File "/home/hummingbot/hummingbot/connector/exchange/kucoin/kucoin_web_utils.py", line 107, in api_request
    raise IOError(f"The request to Kucoin failed. Error: {error_response}. Request: {request}")
OSError: The request to Kucoin failed. Error: {"code":"400005","msg":"Invalid KC-API-SIGN"}. Request: RESTRequest(method=GET, url='https://api.kucoin.com/api/v1/trade-fees', endpoint_url=None, params={'symbols': 'ALGO-USDT,AVAX-USDT,DAO-USDT,FEAR-USDT'}, data=None, headers={'Content-Type': 'application/x-www-form-urlencoded'}, is_auth_required=True, throttler_limit_id='/api/v1/trade-fees')
2022-04-30 14:23:29,906 - 1 - hummingbot.connector.exchange.kucoin.kucoin_exchange - WARNING - Could not fetch new trading fees from Kucoin. Check network connection.
`

**Optional: your discord username:**
Bhagavate bilala#6332

**Comment by nelyajizi** [comment]
have the same problem with hedge strategy

**Comment by aarmoa** [comment]
I was not able to reproduce the issue in version 1.4.0

**Comment by MementoRC** [comment]
This is still happening in v1.4.0 development with the ScriptStrategyBase method.
The issue is that trade-fees does not in fact allow multiple pairs, but just one, this is why it works with the PMM, but not with my ScriptStrategyBase method that has several markets. I changed line 816 of kucoin_exchange.py to:

        params = {"symbols": ",".join(trading_symbols**[:1]**)}

Et voila, no more nagging message and much more elegant than my old fix of hard-coding the fees I have on KuCoin ;P

**Comment by MementoRC** [comment]
The proposal above is not satisfying, it prevents to get fees for the case of multiple pairs that could (not sure) have different fee structure at the exchange, below is a more complete workaround:
        # It appears that the SIGN is incorrect when more than 1 pais is passed
        # This is likely not the correct fix due to the multiplicity of PAI calls
        resp = {'data': []}
        for pair in trading_symbols:
            params = {"symbols": pair}
            r = await self._api_request(
                path_url=CONSTANTS.FEE_PATH_URL,
                params=params,
                method=RESTMethod.GET,
                is_auth_required=True,
            )
            resp['data'] += r['data']

**Comment by cardosofede** [comment]
Thank you @MementoRC, we update the contribution guidelines yesterday.

**Comment by jtraub91** [comment]
Seeing this on `v1.3.0` as well
```
2022-05-19 05:30:33,170 - 1 - hummingbot.connector.exchange.kucoin.kucoin_exchange - WARNING - Could not fetch new trading fees from Kucoin. Check network connection.
2022-05-19 05:30:33,825 - 1 - hummingbot.connector.exchange.kucoin.kucoin_exchange - NETWORK - Unexpected error while fetching trading fees.
Traceback (most recent call last):
  File "/home/hummingbot/hummingbot/connector/exchange/kucoin/kucoin_exchange.py", line 767, in _trading_fees_polling_loop
    await self._update_trading_fees()
  File "/home/hummingbot/hummingbot/connector/exchange/kucoin/kucoin_exchange.py", line 817, in _update_trading_fees
    resp = await self._api_request(
  File "/home/hummingbot/hummingbot/connector/exchange/kucoin/kucoin_exchange.py", line 946, in _api_request
    return await web_utils.api_request(
  File "/home/hummingbot/hummingbot/connector/exchange/kucoin/kucoin_web_utils.py", line 107, in api_request
    raise IOError(f"The request to Kucoin failed. Error: {error_response}. Request: {request}")
OSError: The request to Kucoin failed. Error: {"code":"400005","msg":"Invalid KC-API-SIGN"}. Request: RESTRequest(method=GET, url='https://api.kucoin.com/api/v1/trade-fees', endpoint_url=None, params={'symbols': 'AVAX-USDT,ALGO-USDT,NIM-USDT'}, data=None, headers={'Content-Type': 'application/x-www-form-urlencoded'}, is_auth_required=True, throttler_limit_id='/api/v1/trade-fees')
```

**Comment by jtraub91** [comment]
@MementoRC's aforementioned PR is presumably here: https://github.com/hummingbot/hummingbot/pull/5323

I'm not sure why there are so many changes considering the scope of this issue. Thus, I created a new PR (https://github.com/hummingbot/hummingbot/pull/5332) with only the relevant change. I built from source on Ubuntu20 and did see the exception go away, but have not done any further testing thus far.

Side note, the `development` branch appears to have a regression eliminating my workaround noted [here](https://github.com/hummingbot/hummingbot/issues/5326#issuecomment-1133525658), i.e. using rate_oracle_source of coingecko no longer can retrieve the rate for NIM-USDT

**Comment by MementoRC** [comment]
Initially, the changes were only in the module and it's test but then there
was a isort failure that I was told should be resolved by rebasing the
upstream/development. I must have made a wrong operation. I am brand new to
this process.
I don't understand what you mean by not needing the workaround, I am using
the development version and had to patch it in order to eliminate the
warning
Regards,
Claude

**Comment by jtraub91** [comment]
Hi @MementoRC; apologies

I think you what you needed to do was "rebase" the relevant commits from your branch onto the development branch in order to make sure you include other development changes that have been made in parallel. In general, this should amount to
```
git checkout <your-branch>
git fetch
git rebase -i origin/development
```
and then 
```
git push
```
If you are new to git, I understand it may be a bit confusing, at first.

I would advise against _ever_ doing a `git merge`, unless you are merging changes to master.

At this point, by viewing your [PR](https://github.com/hummingbot/hummingbot/pull/5323), there are so many merge commits it is hard to determine what is relevant.

There are 23 files changed in your PR, but I don't think any of them are related to this issue except the following
- `hummingbot/connector/exchange/kucoin/kucoin_exchange.py`
- `test/hummingbot/connector/exchange/kucoin/test_kucoin_exchange.py`

P.S. I am new to this community as well, happy to chat further here and / or discord :)

**Comment by MementoRC** [comment]
Thanks, I welcome the lesson, I want to use PyCharm and I am not sure what
'git' commands are run with the different menu items, I surely did
something wrong, but I think I did it correctly for the other PR. In fact,
originally this PR looked good with only the changes that you are referring
to, then I selected upstream/development and 'Pull into <my branch> with
Rebase' :(

I should remove this faulty PR now that you have it resubmitted correctly,
shouldn't I?

Thanks, I am MementoMori on Discord

**Comment by globeone** [comment]
Just reinstalled the robot completely fresh. Clean VPS, Ubuntu 22.04 LTS, Fresh Docker container, fresh API key
unfortunately the
Problem still exists in 1.4.0

![image](https://user-images.githubusercontent.com/10052587/171283946-7af9d794-5abd-4948-ace5-80239526b8dc.png)

Here is the output from the Log File. 

`2022-05-31 21:01:27,448 - 1 - hummingbot.connector.exchange.kucoin.kucoin_exchange - NETWORK - Unexpected error while fetching trading fees.
Traceback (most recent call last):
  File "/home/hummingbot/hummingbot/connector/exchange/kucoin/kucoin_exchange.py", line 767, in _trading_fees_polling_loop
    await self._update_trading_fees()
  File "/home/hummingbot/hummingbot/connector/exchange/kucoin/kucoin_exchange.py", line 817, in _update_trading_fees
    resp = await self._api_request(
  File "/home/hummingbot/hummingbot/connector/exchange/kucoin/kucoin_exchange.py", line 946, in _api_request
    return await web_utils.api_request(
  File "/home/hummingbot/hummingbot/connector/exchange/kucoin/kucoin_web_utils.py", line 107, in api_request
    raise IOError(f"The request to Kucoin failed. Error: {error_response}. Request: {request}")
OSError: The request to Kucoin failed. Error: {"code":"400005","msg":"Invalid KC-API-SIGN"}. Request: RESTRequest(method=GET, url='https://api.kucoin.com/api/v1/trade-fees', endpoint_url=None, params={'symbols': 'ALGO-BTC,FRONT-BTC,NIM-BTC,SHR-BTC'}, data=None, headers={'Content-Type': 'application/x-www-form-urlencoded'}, is_auth_required=True, throttler_limit_id='/api/v1/trade-fees')
2022-05-31 21:01:27,448 - 1 - hummingbot.connector.exchange.kucoin.kucoin_exchange - WARNING - Could not fetch new trading fees from Kucoin. Check network connection`


The Status command works and the trades have been set so there are no "network" issues and the API key is valid.

![image](https://user-images.githubusercontent.com/10052587/171284720-69c67eb4-3ec3-41a8-9ebe-fcfbba076dfe.png)

Thanks for having another go at a fix.

**Comment by MementoRC** [comment]
The workaround proposed does not seem to have made it to v1.4.0. This line
is a telltale sign:
params={'symbols': 'ALGO-BTC,FRONT-BTC,NIM-BTC,SHR-BTC'}
